packages/react-reconciler/src/ReactFiberClassComponent.js

```diff
@@ -787,7 +787,10 @@ export default function(
       // Render-phase updates (like this) should not be added to the update queue,
       // So that multiple render passes do not enqueue multiple updates.
       // Instead, just synchronously merge the returned state into the instance.
-      newState = Object.assign({}, newState, partialState);
+      newState =
+        newState == null
```

Comment:
Super-duper nit: strict compare to `null` and `undefined`: `newState === null || newState === undefined`. We don't do this everywhere consistently yet but it's slightly faster.